packages/near-api-js/src/providers/json-rpc-provider.ts

```diff
@@ -149,7 +149,7 @@ export class JsonRpcProvider extends Provider {
         if (result && result.error) {
             throw new TypedError(
                 `Querying failed: ${result.error}.\n${JSON.stringify(result, null, 2)}`,
-                getErrorTypeFromErrorMessage(result.error, result.error.name)
+                getErrorTypeFromErrorMessage(result.error.toString(), result.error.name)
```

Comment:
I think this was happening implicitly when the `/.../.test()` method is invoked with an `Error` object. Since the field being tested is ostensibly the error message, this should be a safe substitute.